### PR TITLE
FIX SS4 compatibility tweaks: namespacing, update readme and Travis config.

### PR DESCRIPTION
**Pull request by robbieaverill**
This pull request contains a few tweaks for SS4 compatibility:

* Implement namespacing (including upgrade map)
* Change namespace from Sminnee\SilverStripeIntercom to SilverStripe\Intercom
* PSR-4 autoloading
* Update readme
* Update Travis configuration for PHP 5.5 to 7.1 builds
* Prevent escaping JSON in template

This does **not** include any of the develop PR code from #1. I'll port the changes through from this PR separately next week since master and develop have diverged a reasonable amount, most notably in the way the Intercom authentication works (API keys are deprecated/removed as of now, and tokens are a requirement).

This also does not rename `code` to `src` because I didn't want to lose Git history on the files. This can be done after merging as a separate PR.

Almost completes #4 (just need to rename `code`)

---

This would require a Packagist update since the package name has changed.

### CODE REVIEW

File: code/IntercomScriptTags.php [hunk 1]
```diff
@@ -1,112 +1,131 @@
...
+     * @return array The settings, ready for JSON-encoding
+     */
+    public function getIntercomSettings(Member $member = null)
+    {
+        if (!$this->isEnabled($member)) {
+            return [];
+        }
+
+        $settings = [
+            'app_id' => INTERCOM_APP_ID,
```
Comment (robbieaverill) [inline code/IntercomScriptTags.php]: We should support a `getenv` call as well

File: code/IntercomScriptTags.php [hunk 2]
```diff
@@ -1,112 +1,131 @@
...
+     */
+    private static $enabled = true;
+
+    /**
+     * @param  Member $member
+     * @return bool
+     */
+    public function isEnabled(Member $member = null)
+    {
+        if (!defined('INTERCOM_APP_ID')) {
```
Comment (robbieaverill) [inline code/IntercomScriptTags.php]: Add support for getenv

File: code/RequestFilter.php [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
-			}
-		}
-	}
+            if ($tags) {
+                $content = $response->getBody();
+                $content = preg_replace("/(<\/body[^>]*>)/i", $tags . "\\1", $content);
+                $response->setBody($content);
+            }
+        }
+    }
```
Comment (robbieaverill) [inline code/RequestFilter.php]: Ideally this class would be replaced with HTTPMiddleware now that RequestFilter is deprecated

File: composer.json [hunk 4]
```diff
@@ -18,5 +18,18 @@
...
             "email": "sam@silverstripe.com"
         }
-    ]
+    ],
+    "autoload": {
+        "psr-4": {
+            "SilverStripe\\Intercom\\": "code/"
+        }
+    },
+    "autoload-dev": {
```
Comment (robbieaverill) [inline composer.json]: Combine these - they shouldn't be separated

File: code/IntercomFormExtension.php [hunk 5]
```diff
@@ -1,226 +1,235 @@
...
+                }
+                $noteData .= '</ul>';
+
+                try {
+                    $intercom->getClient()->notes->create([
+                        'body' => $noteData,
+                        'user' => ['id' => $lead['id']]
+                    ]);
+                } catch (Exception $e) {
+                    Injector::inst()->get('Logger')->addWarning("Could not create note: {$e->getMessage()}");
```
Comment (robbieaverill) [inline code/IntercomFormExtension.php]: Needs to be updated to use `\Psr\Log\LoggerInterface::class` instead of `Logger` now

**Comment by robbieaverill** [comment]
Updated, hopefully the builds will pass now

**Comment by sminnee** [comment]
OK should have a good, green build now. 

@robbieaverill can you have a final look over this and merge if you're happy? Keen to get a 3.0.0-beta1 tagged soon to load onto the in-development ss.com site

**Pull request merged**